### PR TITLE
Simplify comparison processes (and add date_difference)

### PR DESCRIPTION
**Pull request by m-mohr**
This is a proposal to simplify the comparison processes. The discussions in #394 lead me to the conclusion that we may not want to handle date/times in so many processes specifically so that issues can be solved in some central places rather than spreading them into so many processes. Also, the old implementation was a bit suboptimal in that you had to parse the inputs first to detect whether the values are temporal values, potentially making the comparisons slower than needed. Most use-cases can likely be solved with a new process such as `date_difference.`.

This includes a breaking change for the comparison processes. The impact is assumed to be low as most date/time values would usually come from labels in timeseries, but label support seems very limited in back-ends right now. For `between` back-ends can choose to still support the old behavior.

Looking for feedback.

### CODE REVIEW

**Comment by soxofaan** [comment]
makes sense.
as already noted: I don't think this will have practical impact at the moment.

Also note that we already could introduce the new `date_difference` process in the 1.x branch and do the breaking 2.0 changes later

**Comment by m-mohr** [comment]
@soxofaan True, but 1.3 and 2.0 are pretty much 2.0 now, I don't think we'll get to an intermediate 1.3 release honestly, but instead directly start with 2.0-beta/rc.

**Comment by dthiex** [comment]
> I don't think this will have practical impact at the moment.

Can confirm this for our backend as our process implementation (for those) didn't support date formats from the beginning.

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,6 +50,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Removed
 
 - The `examples` folder has been migrated to the [openEO Community Examples](https://github.com/Open-EO/openeo-community-examples/tree/main/processes) repository.
+- `between`: Support for temporal comparison.
```
Comment (dthiex) [inline CHANGELOG.md:53]: Did you think of adding a new process for this? I don't know if it is used/needed by anyone but currently our implementation supports this.

**Review by dthiex — CHANGES_REQUESTED** [review]

**Review by LukeWeidenwalker — APPROVED** [review]
Looks like a good idea to me! Similar to others this will have minimal impact on us - our current implementations don't support comparing datetimes either (but good timing, we're working on those right now and will not spend time on this then!).

File: CHANGELOG.md (hunk 1 above)
Comment (m-mohr) [reply CHANGELOG.md:53]: Not planned right now, but would be something we can add anytime when needed. You can continue to support it.

File: eq.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+    "description": "Compares whether `x` is strictly equal to `y`.\n\n**Remarks:**\n\n* Data types MUST be checked strictly. For example, a string with the content *1* is not equal to the number *1*. Nevertheless, an integer *1* is equal to a floating-point number *1.0* as `integer` is a sub-type of `number`.\n* If any operand is `null`, the return value is `null`.\n* If any operand is an array or object, the return value is `false`.\n* Strings are expected to be encoded in UTF-8 by default.\n* Temporal strings are normal strings. To compare temporal strings use ``date_difference()``.",
```
[the remark this answers is not in the document]
Comment (m-mohr) [reply eq.json:4]:
```suggestion
    "description": "Compares whether `x` is strictly equal to `y`.\n\n**Remarks:**\n\n* Data types MUST be checked strictly. For example, a string with the content *1* is not equal to the number *1*. Nevertheless, an integer *1* is equal to a floating-point number *1.0* as `integer` is a sub-type of `number`.\n* If any operand is `null`, the return value is `null`.\n* If any operand is an array or object, the return value is `false`.\n* Strings are expected to be encoded in UTF-8 by default.\n* Temporal strings are normal strings. To compare temporal strings as dates/times, use ``date_difference()``.",
```

File: proposals/date_difference.json [hunk 3]
```diff
@@ -0,0 +1,99 @@
+{
+    "id": "date_difference",
+    "summary": "Computes the difference between two time instants",
+    "description": "Computes the difference between two instants in time and returns the difference between them in the unit given.\n\nThe process basically converts the given dates into a numerical timestamps and returns the result of subtracting the other date from the base date. If a given date doesn't include the time, the process assumes that the time component is `00:00:00Z` (i.e. midnight, in UTC). The millisecond part of the times are optional and default to `0` if not given. The process doesn't take daylight saving time (DST) into account as only dates and times in UTC (with potential numerical time zone modifier) are supported.",
```
[the remark this answers is not in the document]
Comment (m-mohr) [reply proposals/date_difference.json:4]:
```suggestion
    "description": "Computes the difference between two instants in time and returns the difference between them in the unit given.\n\nThe process converts the given dates into numerical timestamps and returns the result of subtracting the other date from the base date. If a given date doesn't include the time, the process assumes that the time component is `00:00:00Z` (i.e. midnight, in UTC). The millisecond part of the times are optional and default to `0` if not given. The process doesn't take daylight saving time (DST) into account as only dates and times in UTC (with potential numerical time zone modifier) are supported.",
```

File: eq.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (dthiex) [reply eq.json]: I think it would be useful to actually add this to all the processes.

[the remark this answers is not in the document]
Comment (m-mohr) [reply eq.json]: Good point, will do.

**Pull request merged**